dev/com.ibm.ws.jdbc_fat_postgresql/test-applications/postgresqlApp/src/jdbc/fat/postgresql/web/PostgreSQLTestServlet.java

```diff
@@ -404,6 +405,75 @@ public void testPropertyCleanup_schema() throws Exception {
         }
     }
 
+    // Verify PostgreSQL vendor-specific Connection property defaultFetchSize can be set and is reset properly
+    @Test
+    public void testPropertyCleanup_postgre_defaultFetchSize() throws Exception {
+        // Use a DataSource with maxPoolSize=1 and shareable=false so that we always reuse the same underlying connection (if possible)
+        DataSource ds = InitialContext.doLookup("jdbc/postgres/maxPoolSize1");
+
+        int defaultFetchSize = -1;
+        try (Connection con = ds.getConnection()) {
+            verifyClean(con);
+            PGConnection pgCon = con.unwrap(PGConnection.class);
+            defaultFetchSize = pgCon.getDefaultFetchSize();
+
+            assertTrue("Was expecting default value to be anything other than 5", defaultFetchSize != 5);
+            pgCon.setDefaultFetchSize(5);
+            assertEquals(5, pgCon.getDefaultFetchSize());
+        }
+        try (Connection con = ds.getConnection()) {
+            verifyClean(con);
```

Comment:
I know that 'verifyClean' was written as part of this pull, however, the test case uses it, and so I was curious what it was doing.  I see this at the start of it which seems very suspicious,
```
        // Always "do some work" with the connection before we verify it's underlying state.
        // The JDBC code intentionally lazily resets connection values, so our wrapper may be out of sync with the underlying
        // connection between getting the initial connection and actually driving some work on it
        con.createStatement().close();
```
Are we covering up a bug where our connection wrapper reports wrong values until someone invokes `createStatement`?  If so, that seems like a serious problem.  For example, imagine if the user has code like this - they could run under the wrong isolation level,
```
if (con.getTransactionIsolation() != Connection.TRANSACTION_REPEATABLE_READ)
   con.setTransactionIsolation(Connection.TRANSACTION_REPEATABLE_READ);
con.createStatement().execute(sql) ...
```

Reply:
no, I don't think we are covering up a bug here. We have an optimization in the JDBC code (which causes more complexity than it was probably worth IMO) where we intentionally let our connection handle and the underlying connection impl be out of sync until the user attempts to do something meaningful with the connection. 

The connection handle that the user app deals with always has correct values, it's the underlying impl object which may have "wrong" values at first, but we correct this lazily as the connection is used. In the code example you gave with setting the tx isolation level, it would play out like this:
1. handle has isolation=RR, underlying impl may have something else (let's say SER)
2. when we attempt to create a statement, we first align connection properties by calling setTxIsolation(RR) on the underlying connection
3. statement is created/executed with RR isolation level

Reply:
Because verify clean does a con.createStatement, which syncs up values, it is depriving the test of actually checking the value on the connection handle prior to the sync.  If it is included, it should be moved further down in the test until after the value of the handle is checked.  Also, the createStatement within verifyClean should probably be moved further down within that method until after the values of the handle are checked.